app/src/main/java/com/example/campusguide/directions/GetDirectionsDialogFragment.kt

```diff
@@ -52,11 +54,14 @@ class GetDirectionsDialogFragment constructor(private val options: DirectionsDia
                     val start = startEditText?.text.toString()
                     val end = endEditText?.text.toString()
 
-                    options.confirmationListener.onConfirm(start, end)
+                    options.confirmationListener.onConfirm(start, end, selectedTravelMode)
                 }
                 .setNegativeButton("Cancel") { dialog, _ ->
                     dialog.cancel()
                 }
+                .setSingleChoiceItems(R.array.travel_modes, -1, DialogInterface.OnClickListener { _, which ->
```

Comment:
Would it be possible to make this allow selecting multiple options? For someone who might want walking or bus, but not car